spec/unit/classes/master/config_spec.rb

```diff
@@ -1,99 +1,93 @@
 require 'spec_helper'
 
 describe 'puppetdb::master::config', :type => :class do
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) do
+        facts.merge({
+            :puppetversion => Puppet.version,
+            :fqdn => 'puppetdb.example.com',
```

Comment:
instead of mocking the fqdn fact you can set the node: https://github.com/voxpupuli/puppet-zabbix/blob/master/spec/classes/agent_spec.rb#L4-L6